ospfd/ospf_opaque.c

```diff
@@ -1314,6 +1323,16 @@ void ospf_opaque_lsa_dump(struct stream *s, uint16_t length)
 	struct ospf_lsa lsa = {};
 
 	lsa.data = (struct lsa_header *)stream_pnt(s);
+	/*
+	 * Check if lsa.data is NULL to prevent null pointer
+	 * dereferencing in show_opaque_info_detail
+	 * when parsing the LSA.
+	 */
+	if (!lsa.data) {
```

Comment:
hmm - could this check just be part of the checking in 'show_opaque_info_detail' itself? that'd make the 'show' function more robust, and sort of confine the validation.